test/src/kbasesearchengine/test/main/IndexerWorkerTest.java

```diff
@@ -1050,13 +1051,273 @@ public SourceData answer(final InvocationOnMock inv) throws Throwable {
                         "GUID code:4/5/6 not found"))));
         
         verify(idxStore, never()).indexObjects(
-                any(), any(), any(), any(), any(), any(), anyBoolean());
+                any(), any(), any(), any(), any(), any());
         
         verify(storage).store(eq(event), eq("GUID_NOT_FOUND"), argThat(new ThrowableMatcher(
                 new UnprocessableEventIndexingException(ErrorType.GUID_NOT_FOUND,
                         "GUID code:4/5/6 not found"))));
     }
-    
+
+    @Test
+    public void indexDependentGUIDPublic() throws Exception {
+        /* tests the checking if the dependent object is indexed with the right permission. */
+        final GUID guid = new GUID("code:1/2/3");
+        final GUID dependencyGUID = new GUID("code:4/5/6");
+        final SearchObjectType dependentType = new SearchObjectType("Assembly", 1);
+        final Map<String, Object> data = ImmutableMap.of("assy_ref", dependencyGUID.toString());
+
+        final EventHandler ws = mock(EventHandler.class);
+        final StatusEventStorage storage = mock(StatusEventStorage.class);
+        final IndexingStorage idxStore = mock(IndexingStorage.class);
+        final TypeStorage typeStore = mock(TypeStorage.class);
+        final LineLogger logger = mock(LineLogger.class);
+        final kbasesearchengine.search.PostProcessing pp = mock(kbasesearchengine.search.PostProcessing.class);
+
+        final Path tempDir = Paths.get(TestCommon.getTempDir()).toAbsolutePath()
+                .resolve("IndexerWorkerTest");
+        deleteRecursively(tempDir);
+
+        when(ws.getStorageCode()).thenReturn("code");
+
+        final IndexerWorkerConfigurator.Builder wrkCfg = IndexerWorkerConfigurator.getBuilder(
+                "myid", tempDir, logger)
+                .withStorage(storage, typeStore, idxStore)
+                .withEventHandler(ws);
+
+        final IndexerWorker worker = new IndexerWorker(wrkCfg.build());
+
+        when(idxStore.checkParentGuidsExist(set(guid))).thenReturn(ImmutableMap.of(guid, false));
+
+        when(ws.load(eq(Arrays.asList(guid)), any(Path.class)))
+                .thenAnswer(new Answer<SourceData>() {
+
+                    @Override
+                    public SourceData answer(final InvocationOnMock inv) throws Throwable {
+                        final Path path = inv.getArgument(1);
+                        new ObjectMapper().writeValue(path.toFile(), data);
+                        return SourceData.getBuilder(
+                                new UObject(path.toFile()), "myobj", "somedude")
+                                .withNullableMD5("md5")
+                                .withIsPublic(false)
```

Comment:
Would help speed up review if you add comments next to the main components of the test. 

.withIsPublic(false)   // set to false

Reply:
Done